### PR TITLE
🌸✨ `Marketplace`:  Begin to smooth out Onboarding

### PR DESCRIPTION
**Pull request by zspencer**
- https://github.com/zinc-collective/convene/issues/1641

This follows the patterns @KellyAH set in #1585 to:

- Enumerate each action that wants to be taken to complete Onboarding for the `Marketplace`
- Shows the warnings in the `ManagementComponent`, so they're everywhere!

## After

Screenshot 2023-07-05 at 8 01 45 PM

## Before

Screenshot 2023-07-05 at 8 05 00 PM

### CODE REVIEW

File: app/furniture/marketplace/onboarding_component.html.erb
Comment (daltonrpruitt) [inline app/furniture/marketplace/onboarding_component.html.erb:1]: Non-blocking, but just a note. 
For some reason, the GitHub UI is not coloring the code correctly. It is doing that for the added ruby file below, but not this added html.erb file?

Comment (zspencer) [reply app/furniture/marketplace/onboarding_component.html.erb:1]: That is very weird; it's coloring other `html.erb` files but not this one...

File: spec/furniture/marketplace/onboarding_component_spec.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  it { is_expected.to have_link(t("marketplace.marketplace.edit.link_to"), href: polymorphic_path(marketplace.location(:edit))) }
+
+  it { is_expected.to have_content(t(".missing.notification_methods", link: t("marketplace.notification_methods.new.link_to"))) }
+  it { is_expected.to have_link(t("marketplace.notification_methods.new.link_to"), href: polymorphic_path(marketplace.location(:new, child: :notification_method))) }
+
+  it { is_expected.to have_content(t(".missing.products", link: t("marketplace.products.new.link_to"))) }
+  it { is_expected.to have_link(t("marketplace.products.new.link_to"), href: polymorphic_path(marketplace.location(:new, child: :product))) }
+
+  it { is_expected.to have_content(t(".missing.delivery_areas", link: t("marketplace.delivery_areas.new.link_to"))) }
+  it { is_expected.to have_link(t("marketplace.delivery_areas.new.link_to"), href: polymorphic_path(marketplace.location(:new, child: :delivery_area))) }
```
Comment (daltonrpruitt) [inline spec/furniture/marketplace/onboarding_component_spec.rb:24]: This is the "first half of the happy-path" towards getting people to add the related items to their marketplace. This still needs the second half at some point.

Comment (zspencer) [reply spec/furniture/marketplace/onboarding_component_spec.rb:24]: Great point, will add the "What happens when it's fully configured? side of things.

**Review by daltonrpruitt — APPROVED** [review]

**Pull request merged**